### PR TITLE
e2e tests for favourite directive

### PR DESCRIPTION
**Pull request by gmandakini**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: e2e/content-services/directives/favourite-directive.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        done();
+    });
+
+    beforeEach(async (done) => {
+        navigationBarPage.clickContentServicesButton();
+        contentServicesPage.getContentList().dataTablePage().waitTillContentLoaded();
+        done();
+    });
+
+    it('[C260247] Favorite a file', async () => {
```
Comment (marouanbentaleb) [inline e2e/content-services/directives/favourite-directive.e2e.ts]: Can we change the name of the tests to follow the convention ? It should be something like "Should be able to mark a file as favourite". The name needs to be changed also in the test case.

File: e2e/content-services/directives/favourite-directive.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+    beforeAll(async (done) => {
+
+        await this.alfrescoJsApi.login(browser.params.testConfig.adf.adminEmail, browser.params.testConfig.adf.adminPassword);
+
+        await this.alfrescoJsApi.core.peopleApi.addPerson(acsUser);
+
+        await this.alfrescoJsApi.login(acsUser.id, acsUser.password);
+
+        testFolder1 = await uploadActions.createFolder(StringUtil.generateRandomString(5), '-my-');
```
Comment (eromano) [inline e2e/content-services/directives/favourite-directive.e2e.ts]: can you please refactor those spaces?

File: demo-shell/src/app/components/files/files.component.html [hunk 3]
```diff
@@ -121,7 +121,9 @@
                             matTooltip="{{ 'DOCUMENT_LIST.TOOLBAR.DELETE' | translate }}">
                         <mat-icon>delete</mat-icon>
                     </button>
-                    <button mat-icon-button
+                    <button
+                        data-automation-id="favourite"
```
Comment (adinapitul) [inline demo-shell/src/app/components/files/files.component.html]: Please use the american english form: favorite

File: e2e/content-services/directives/favourite-directive.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+import { AcsUserModel } from '../../models/ACS/acsUserModel';
+import { AlfrescoApiCompatibility as AlfrescoApi } from '@alfresco/js-api';
+import { browser } from 'protractor';
+import { FileModel } from '../../models/ACS/fileModel';
+import resources = require('../../util/resources');
+import { NavigationBarPage } from '../../pages/adf/navigationBarPage';
+import { CustomSources } from '../../pages/adf/demo-shell/customSourcesPage';
+import { TrashcanPage } from '../../pages/adf/trashcanPage';
+
+describe('Favourite directive', function () {
```
Comment (adinapitul) [inline e2e/content-services/directives/favourite-directive.e2e.ts]: Favorite

File: e2e/content-services/directives/favourite-directive.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+        contentServicesPage.getContentList().dataTablePage().selectRow('Display name', testFolder1.entry.name);
+        contentServicesPage.getContentList().dataTablePage().checkRowIsSelected('Display name', testFolder1.entry.name);
+        contentServicesPage.clickOnFavouriteButton();
+        contentServicesPage.checkIsNotMarkedFavourite();
+        customSourcesPage.navigateToCustomSources();
+        customSourcesPage.selectFavouritesSourceType();
+        customSourcesPage.checkRowIsNotDisplayed(testFolder1.entry.name);
+    });
+
+    it('[C260251] Should retain the restored file as favourite', async () => {
```
Comment (adinapitul) [inline e2e/content-services/directives/favourite-directive.e2e.ts]: This has nothing to do with the favorite directive

File: e2e/content-services/directives/favourite-directive.e2e.ts [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+        contentServicesPage.getContentList().dataTablePage().checkContentIsDisplayed('Display name', testFile.entry.name);
+        contentServicesPage.getContentList().dataTablePage().selectRow('Display name', testFile.entry.name);
+        contentServicesPage.getContentList().dataTablePage().checkRowIsSelected('Display name', testFile.entry.name);
+        contentServicesPage.checkIsMarkedFavourite();
+        customSourcesPage.navigateToCustomSources();
+        customSourcesPage.selectFavouritesSourceType();
+        customSourcesPage.checkRowIsDisplayed(testFile.entry.name);
+    });
+
+    it('[C260252] Should retain the moved file as favourite', async () => {
```
Comment (adinapitul) [inline e2e/content-services/directives/favourite-directive.e2e.ts]: Nothing to do with the favorite directive

**Comment by adinapitul** [comment]
Please use american English form : favorite instead of favourite, everywhere

**Pull request merged**